src/main/java/tech/strategio/tests/ShoppingCartTest.java

```diff
@@ -0,0 +1,105 @@
+package tech.strategio.tests;
+
+import tech.strategio.Product;
+import tech.strategio.ProductNotFoundException;
+import tech.strategio.ShoppingCart;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.*;
+
+/*
+When created, the cart has 0 items
+When empty, the cart has 0 items
+When a new product is added, the number of items must be incremented
+When a new product is added, the new balance must be the sum of the previous balance plus the cost of the new product
+When an item is removed, the number of items must be decreased
+When a product not in the cart is removed, a ProductNotFoundException must be thrown
+Hint: insert the call in a try block and put a fail() after the call to removeItem()
+*/
+
+class ShoppingCartTest {
+    private ShoppingCart cart;
+
+    @BeforeEach
+    void setUp() {
+        cart = new ShoppingCart();
+    }
+
+    @Test
+        // When created, the cart has 0 items
+    void test_new_cart_is_empty() {
+        assertEquals(0, cart.getItemCount());
+    }
+
+    @Test
+        // When empty, the cart has 0 items
+    void test_empty_cart_has_0_items() {
+        cart.addItem(new Product("car", 42345));
+        assertEquals(1, cart.getItemCount());
+        cart.empty();
+        assertEquals(0, cart.getItemCount());
+    }
+
+    @Test
+        // When a new product is added, the number of items must be incremented
+    void test_if_item_added_to_cart_increments() {
+        cart.addItem(new Product("Celery", 4.99));
+        assertEquals(1, cart.getItemCount());
+        cart.addItem(new Product("Cheese", 2.49));
+        assertEquals(2, cart.getItemCount());
+        cart.addItem(new Product("Banana", 6.09));
+        assertEquals(3, cart.getItemCount());
+    }
+
+    @Test
+        // When a new product is added, the new balance must be the sum
+        // of the previous balance plus the cost of the new product
+    void test_new_product_added_balance() {
+//        Alternate implementation
```

Comment:
So usually, it's best practice to never leave commented code inside of a pull request.